clients/client-acm-pca/waiters/waitForAuditReportCreated.ts

```diff
@@ -42,3 +47,19 @@ export const waitForAuditReportCreated = async (
   const serviceDefaults = { minDelay: 3, maxDelay: 120 };
   return createWaiter({ ...serviceDefaults, ...params }, input, checkState);
 };
+/**
+ * Wait until a Audit Report is created
+ *  @param params : Waiter configuration options.
+ *  @param input : the input to DescribeCertificateAuthorityAuditReportCommand for polling.
+ */
+export const waitUntilAuditReportCreated = async (
+  params: WaiterConfiguration<ACMPCAClient>,
+  input: DescribeCertificateAuthorityAuditReportCommandInput
+): Promise<WaiterResult> => {
+  const serviceDefaults = { minDelay: 3, maxDelay: 120 };
+  const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState);
+  if (result.state != WaiterState.SUCCESS) {
+    throw result;
```

Comment:
Throw an error, and not an object.
```suggestion
    throw Object.assign(new Error(result.state), result.reason);
```

Recommendation in codegen https://github.com/awslabs/smithy-typescript/pull/318#discussion_r622649595

Reply:
There's new recommendation on throwing `"AbortError"` in case of `ABORTED` state https://github.com/awslabs/smithy-typescript/pull/318#discussion_r623272932

Reply:
handled